FakeServer/Controllers/DynamicController.cs

```diff
@@ -147,6 +158,21 @@ private IActionResult GetCollectionItem(string collectionId, int skip, int take)
                 results = SortHelper.SortFields(results, options.SortFields);
             }
 
+            if (_cachingSettings.ETag.Enabled)
+            {
+                var content = JsonConvert.SerializeObject(results);
```

Comment:
Move functionality to ETag-class, as this needs to be added also to AsyncController. Could you also add some test for the AsyncController?


GenerateEtag could take dynamic results as parameter and return etag.

```cs
if (_cachingSettings.ETag.Enabled)
{
	
	var eTag = ETag.GenerateETag(results);
	Response.Headers[HeaderNames.ETag] = eTag;

	if (ETag.IsETagSameInRequest(Request, eTag))
		return StatusCode((int)HttpStatusCode.NotModified);
}
```